File: sem4/mpp/mpp-proiect-csharp-vreauladudu/persistence/repository/database/MotorcycleDatabase.cs
using mpp_proiect_csharp_vreauladudu.domain;
using System;
using System.Data.SQLite;
using System.Collections.Generic;

namespace mpp_proiect_csharp_vreauladudu.repository.database
{
    public class MotorcycleDatabase : AbstractDatabase<int, Motorcycle>, MotorcycleRepository
    {
        public MotorcycleDatabase() : base("Motorcycles") { }

        protected override Motorcycle extractEntity(SQLiteDataReader result)
        {   
            int id = Convert.ToInt32(result["id"]);
            String brand = Convert.ToString(result["brand"]);
            int capacity = Convert.ToInt32(result["capacity"]);

            Motorcycle motorcycle = new Motorcycle();
            motorcycle.Id = id;
            motorcycle.Brand = brand;
            motorcycle.Capacity = capacity;
            return motorcycle;
        }

        protected override SQLiteCommand findStatement(int id)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "SELECT * FROM Motorcycles WHERE [id] = @id";
            command.Parameters.Add(new SQLiteParameter("@id", id));
            return command;
        }

        protected override SQLiteCommand saveStatement(Motorcycle motorcycle)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "INSERT INTO Motorcycles(id, brand, capacity) VALUES (@id, @brand, @capacity)";
            command.Parameters.Add(new SQLiteParameter("@id", motorcycle.Id));
            command.Parameters.Add(new SQLiteParameter("@brand", motorcycle.Brand));
            command.Parameters.Add(new SQLiteParameter("@capacity", motorcycle.Capacity));
            return command;
        }

        protected override SQLiteCommand deleteStatement(int id)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "DELETE FROM Motorcycles WHERE [id] = @id";
            command.Parameters.Add(new SQLiteParameter("@id", id));
            return command;
        }

        protected override SQLiteCommand updateStatement(Motorcycle motorcycle)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "UPDATE Motorcycles SET [brand] = @brand, [capacity] = @capacity WHERE [id] = @id";
            command.Parameters.Add(new SQLiteParameter("@brand", motorcycle.Brand));
            command.Parameters.Add(new SQLiteParameter("@capacity", motorcycle.Capacity));
            command.Parameters.Add(new SQLiteParameter("@id", motorcycle.Id));
            return command;
        }

        public List<Motorcycle> findByCapacity(int Capacity)
        {
            List<Motorcycle> motorcycles = new List<Motorcycle>();

            using (SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand())
            {
                command.CommandText = "SELECT * from Motorcycles WHERE [capacity] = @capacity";
                command.Parameters.Add(new SQLiteParameter("@capacity", Capacity));
                using (SQLiteDataReader result = command.ExecuteReader())
                {
                    while (result.Read())
                    {
                        motorcycles.Add(this.extractEntity(result));
                    }
                    return motorcycles;
                }
            }
        }

        public List<Motorcycle> findByBrand(String Brand)
        {
            List<Motorcycle> motorcycles = new List<Motorcycle>();

            using (SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand())
            {
                command.CommandText = "SELECT * from Motorcycles WHERE [brand] = @brand";
                command.Parameters.Add(new SQLiteParameter("@brand", Brand));

                using (SQLiteDataReader result = command.ExecuteReader())
                {
                    while (result.Read())
                    {
                        motorcycles.Add(this.extractEntity(result));
                    }
                    return motorcycles;
                }
            }
        }

        public Motorcycle findById(int Id)
        {
            using (SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand())
            {
                command.CommandText = "SELECT * from Motorcycles WHERE [id] = @id";
                command.Parameters.Add(new SQLiteParameter("@id", Id));

                using (SQLiteDataReader result = command.ExecuteReader())
                {
                    return result.Read() ? this.extractEntity(result) : null;
                }
            }
        }
    }
}

File: sem4/mpp/mpp-proiect-java-vreauladudu/persistence/src/main/java/moto/repository/database/MotorcycleDatabase.java
package moto.repository.database;

import moto.Motorcycle;
import moto.repository.MotorcycleRepository;
import org.springframework.stereotype.Component;

import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

@Component
public class MotorcycleDatabase extends AbstractDatabase<Integer, Motorcycle> implements MotorcycleRepository
{
    public MotorcycleDatabase(Properties properties)
    {
        super(properties, "Motorcycles");
    }

    @Override
    protected Motorcycle extractEntity(ResultSet resultSet) throws SQLException
    {
        Integer id = resultSet.getInt("id");
        String brand = resultSet.getString("brand");
        Integer capacity = resultSet.getInt("capacity");

        Motorcycle motorcycle = new Motorcycle();
        motorcycle.setId(id);
        motorcycle.setBrand(brand);
        motorcycle.setCapacity(capacity);
        return motorcycle;
    }

    @Override
    protected PreparedStatement findStatement(Integer id) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("SELECT * from Motorcycles WHERE id = ?");
        statement.setInt(1, id);
        return statement;
    }

    @Override
    protected PreparedStatement saveStatement(Motorcycle motorcycle) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("INSERT INTO Motorcycles(id, brand, capacity) VALUES (?, ?, ?)");
        statement.setInt(1, motorcycle.getId());
        statement.setString(2, motorcycle.getBrand());
        statement.setInt(3, motorcycle.getCapacity());
        return statement;
    }

    @Override
    protected PreparedStatement deleteStatement(Integer id) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("DELETE FROM Motorcycles WHERE id = ?");
        statement.setInt(1, id);
        return statement;
    }

    @Override
    protected PreparedStatement updateStatement(Motorcycle motorcycle) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("UPDATE Motorcycles SET brand = ?, capacity = ? WHERE id = ?");
        statement.setString(1, motorcycle.getBrand());
        statement.setInt(2, motorcycle.getCapacity());
        statement.setInt(3, motorcycle.getId());
        return statement;
    }

    @Override
    public List<Motorcycle> findByCapacity(Integer Capacity) throws SQLException
    {
        List<Motorcycle> motorcycles = new ArrayList<>();

        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("SELECT * from Motorcycles WHERE capacity = ?");
        statement.setInt(1, Capacity);
        ResultSet resultSet = statement.executeQuery();
        while (resultSet.next())
        {
            motorcycles.add(this.extractEntity(resultSet));
        }
        return motorcycles;
    }

    @Override
    public List<Motorcycle> findByBrand(String Brand) throws SQLException
    {
        List<Motorcycle> motorcycles = new ArrayList<>();

        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("SELECT * from Motorcycles WHERE brand = ?");
        statement.setString(1, Brand);
        ResultSet resultSet = statement.executeQuery();
        while (resultSet.next())
        {
            motorcycles.add(this.extractEntity(resultSet));
        }
        return motorcycles;
    }

    @Override
    public Motorcycle findById(Integer Id) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("SELECT * from Motorcycles WHERE id = ?");
        statement.setInt(1, Id);
        ResultSet resultSet = statement.executeQuery();
        return resultSet.next() ? this.extractEntity(resultSet) : null;
    }
}